src/librustc_index/vec.rs

```diff
@@ -204,39 +204,39 @@ macro_rules! newtype_index {
             #[inline]
             fn steps_between(start: &Self, end: &Self) -> Option<usize> {
                 <usize as ::std::iter::Step>::steps_between(
-                    &Idx::index(*start),
-                    &Idx::index(*end),
+                    &Self::index(*start),
+                    &Self::index(*start),
```

Comment:
end => start here in the diff intentional?

r=me once confirmed that it is or changed if not.

Reply:
This should be `end`, yes.